lib/models/depot.dart

```diff
@@ -47,6 +49,23 @@ class Depot {
     });
   }
 
+  // A method to the filter depot list
+  static Future<List<Depot>> depotFilter(dynamic database, String text) async {
+    final db = await database;
+
+    //Query
+    final List<Map<String, dynamic>> maps =
+        await db.query('depot', where: 'text LIKE ?', whereArgs: ['%$text%']);
+
+    // Convert the List<Map<String, dynamic> into a List<Depot>.
+    return List.generate(maps.length, (i) {
+      return Depot(
+        uuid: maps[i]['uuid'],
+        text: maps[i]['text'],
+      );
+    });
+  }
```

Comment:
Does this function handle empty text? because we assume that if there is an empty text it must returns all depots, I guess.